### DISCUSSION TITLE
OpenCL VDF Verification Times for various hardware

### DISCUSSION THREAD

**Discussion by scrutinously** [madMAx43v3r/mmx-node#37] [General]
It may help to have a list of reference hardware to compare different options. Please include the network version the time was on, the opencl device and drivers, and if possible include the system power consumption seen when running.

Testnet4
Nvidia quadro k420, nvidia 470.xx drivers
Average VDF time 2.47s
Power draw: 110w, peaking at 140w for over 1s when verifying VDF

Testnet4
Nvidia quadro k620, nvidia 510.xx drivers
Average VDF time 0.3s
Power draw 110w, peaking at 125w for an instant when verifying VDF

Testnet4
ATI Firepro v3900, Mesa OpenCL drivers on linux:
Average VDF time 2.97s
Unfortunately I did not write down the power draw I saw with this, but I remember it not being very good.

Testnet3
ATI Firepro v3900, windows catalyst drivers 15.201.2401
Average VDF time ~1s

I should add, my power measurement isn't the most scientific, I simply watch my killawatt and see that it settles out at one number most of the time, then jumps up every 10s which corresponds to VDF timing.

**Comment by Ealrann** [comment]
Testnet4
MSI RADEON RX 560 AERO ITX 4G
Driver: opencl-amd 21.40.2.1350682-3 (archlinux)
Average VDF time: 0.3s
Power draw: 90w (my power consumption meter is too slow to see any peak).

**Comment by madMAx43v3r** [comment]
Quadro K2000 on Ubuntu 18.04 with 470 driver:
VDF Average = 1.44778 sec (at ~32000 tdiff)

Power draw hard to say, server does a lot of jobs with almost all PCIe filled with cards, plus 8x 3.5 inch HDDs, around 250W.

**Comment by madMAx43v3r** [comment]
Intel i7-9750H with UHD 630 on Ubuntu 20.40: 
VDF Average = 0.971067 sec (at ~32000 tdiff)

**Comment by xkredr** [comment]
Intel® UHD Graphics 600 passed through to a Ubuntu20.04 VM (xcp-NG hypervisor running on an Intel NUC7CJYH with Celeron J4005)
intel-opencl-icd
VDF Average = 3.11 sec (tdiff ~32250)
System Power ~10-15W depending on VDF activity

testnet4

**Comment by rhernaus** [comment]
Testnet4
Ubuntu 21.10, kernel 5.13
Nvidia GTX 1080 Ti, Nvidia 510.xx drivers
Average VDF time 0.12s
GPU TDP: 250W

Testnet4
Ubuntu 20.04, kernel 5.13
Nvidia GT 710, Nvidia 470.xx drivers
Average VDF time 1.45s
GPU TDP: 19W

Testnet4
Raspbian Bullseye, kernel 5.10
Raspberry Pi 4, CPU
Average VDF time ~30s. <10s is needed to keep up with the blockchain.

**Comment by xkredr** [comment]
Intel® Iris® Plus Graphics 655 passed through to a Ubuntu20.04 VM (xcp-NG hypervisor running on an Intel NUC8i5BEH  with i5-8259U)
intel-opencl-icd
VDF Average = 1.30 sec (tdiff ~35250)
System Power ~15-35W depending on VDF activity (also running a VM with Chia full-node)

testnet4

**Comment by xkredr** [comment]
Nvidia Quadro K2200 on Ubuntu 20.04, v.470 driver
VDF Average = 0.26s (tdiff ~35100)
System power ~45-70W depending on VDF activity, so 25W for the K2200 at work.

testnet4

**Reply by scrutinously** [reply]
The k2200 is a maxwell chip  like my k620, so It should work on the current nvidia driver and not require the 470 driver like kepler chips do. Interesting results that the k2200 (and even k620) are so close in performance to the T600. Makes sense with the same core count and only slightly higher fp32 performance compared to the k2200.

**Comment by xkredr** [comment]
Nvidia T600 on Ubuntu 20.04, v510 driver
VDF Average = 0.22s (tdiff 35000)
No power measurement but since it uses 12nm technology it should be (much) more efficient than f.i. the K2200 with 28nm tech.

testnet4

**Comment by QuintLeo** [comment]
Testnet 12
PNY physical 8-bit interface low profile Nvidea GT 1030 (MadMax's recommended type of card).
TDP 19 watts but seems to use half that or less (nvidia-smi shows "n/a" / 19 in the power display area).
52 degrees when a VCF is running, 46-48 rest of the time - in a system that's fairly warm right now due to our current heat wave.
v535.xx driver.
VDF 2.1-2.2 seconds (WAY less than the <4 seconds he specifies)
Linux Mint 21.3 Kernel 5.15(?)
